src/lxc/tools/lxc_unshare.c

```diff
@@ -112,6 +117,14 @@ static int do_start(void *arg)
 	int want_default_mounts = start_arg->want_default_mounts;
 	const char *want_hostname = start_arg->want_hostname;
 
+	if (start_arg->setuid) {
+		/* waiting until uid maps is set */
+		if (read(wait_fd, &val, sizeof(val)) == -1) {
```

Comment:
Nit, please use:

    ret = read();
    if (ret < 0) {
            /* stuff happening */
    }

as we do it all over the codebase.

Reply:
And please close the wait_fd on error conditions. It's just good practice.